File: veille_matin.py
"""
KAIROS — Veille IA du matin (8h)
Script one-shot déclenché par Windows Task Scheduler
"""

import subprocess
import requests
from datetime import datetime
from pathlib import Path

BOT_TOKEN = "8881380387:AAHPQEUuZSptb5UFNXq0ZsctM6lNKc6SrCE"
CHAT_ID   = 7579717556
BOT_DIR   = Path(r"C:\Users\chapo\Documents\Agent Marketing V1")

today = datetime.now().strftime("%A %d %B %Y")

PROMPT = f"""veille

Génère le rapport de veille IA du matin pour aujourd'hui ({today}).

Format Telegram (Markdown avec *gras* et bullets). Structure exacte :

*Veille IA — {datetime.now().strftime("%d/%m")}*

*Claude / Anthropic*
[2-3 points sur Claude, Anthropic, nouvelles capacités ou actus récentes]

*Tendances contenu IA*
[2-3 sujets viraux ou tendances sur TikTok/YouTube liés à l'IA]

*Idées de scripts cette semaine*
• [Titre accrocheur #1] — [pourquoi c'est viral en 1 ligne]
• [Titre accrocheur #2] — [pourquoi c'est viral en 1 ligne]
• [Titre accrocheur #3] — [pourquoi c'est viral en 1 ligne]

*Meilleur sujet du jour : [X]/10*
[Une ligne de justification]

Sois concis. Actionnable. Sujets que je peux tourner aujourd'hui."""

def main():
    print(f"[{datetime.now().strftime('%H:%M:%S')}] Génération veille matin...")

    try:
        result = subprocess.run(
            ["claude", "-p", PROMPT],
            capture_output=True,
            text=True,
            cwd=str(BOT_DIR),
            timeout=120,
            encoding="utf-8"
        )
        text = (result.stdout or "").strip()
        if not text:
            text = "Erreur génération veille — relance manuellement avec `veille`."
    except Exception as e:
        text = f"Erreur veille : {e}"

    # Envoyer sur Telegram
    requests.post(
        f"https://api.telegram.org/bot{BOT_TOKEN}/sendMessage",
        json={
            "chat_id": CHAT_ID,
            "text": text,
            "parse_mode": "Markdown",
            "disable_web_page_preview": True
        },
        timeout=15
    )
    print("Veille envoyée.")

if __name__ == "__main__":
    main()

File: veille_matin.mjs
/**
 * KAIROS — Veille IA du matin
 * Déclenché par Windows Task Scheduler à 8h
 * node veille_matin.mjs
 */

import { execFile } from "child_process";
import { promisify } from "util";

const execFileAsync = promisify(execFile);

const BOT_TOKEN = "8881380387:AAHPQEUuZSptb5UFNXq0ZsctM6lNKc6SrCE";
const CHAT_ID   = 7579717556;
const BOT_DIR   = "C:\\Users\\chapo\\Documents\\Agent Marketing V1";

const today = new Date().toLocaleDateString("fr-FR", {
  weekday: "long", day: "numeric", month: "long", year: "numeric"
});
const todayShort = new Date().toLocaleDateString("fr-FR", { day: "2-digit", month: "2-digit" });

const PROMPT = `veille

Génère le rapport de veille IA du matin. Date : ${today}.

Format Telegram Markdown (*gras*, bullets •). Structure exacte :

*Veille IA — ${todayShort}*

*Claude / Anthropic*
[2-3 points sur Claude, Anthropic, nouvelles capacités ou actus]

*Tendances contenu IA*
[2-3 sujets viraux ou tendances TikTok/YouTube liés à l'IA]

*Idées de scripts cette semaine*
• [Titre accrocheur] — [pourquoi viral en 1 ligne]
• [Titre accrocheur] — [pourquoi viral en 1 ligne]
• [Titre accrocheur] — [pourquoi viral en 1 ligne]

*Meilleur sujet du jour : X/10*
[Justification en 1 ligne]

Concis. Actionnable. Sujets que je peux tourner aujourd'hui.`;

async function main() {
  console.log("Génération veille matin...");
  let text;

  try {
    const { stdout, stderr } = await execFileAsync(
      "claude", ["-p", PROMPT],
      { cwd: BOT_DIR, timeout: 120_000, maxBuffer: 1024 * 1024 * 5 }
    );
    text = (stdout || stderr || "").trim() || "Erreur — génère manuellement avec `veille`.";
  } catch (err) {
    text = `Erreur veille : ${err.message}`;
  }

  const res = await fetch(`https://api.telegram.org/bot${BOT_TOKEN}/sendMessage`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      chat_id: CHAT_ID,
      text,
      parse_mode: "Markdown",
      disable_web_page_preview: true,
    }),
  });

  const data = await res.json();
  console.log(data.ok ? "Veille envoyée ✅" : `Erreur Telegram: ${data.description}`);
}

main().catch(console.error);
